benchmark/net/chunky_http_client.js

```diff
@@ -0,0 +1,104 @@
+"use strict";
+// test HTTP throughput in fragmented header case
+var common = require('../common.js');
+var net = require('net');
+var test = require('../../test/common.js');
+
+var bench = common.createBenchmark(main, {
+//  len: [1, 4, 8, 16, 32, 64, 128],
+  len: [1],
+  num: [5, 50, 500, 2000],
+  type: ['send'],
+}
+)
+
+
+function main(conf) {
+  var len;
+  var num;
+  var type;
+  num = +conf.num;
+  len = +conf.len;
+  type = conf.type;
+  var todo = [];
+  var headers = [];
+  // Chose 7 because 9 showed "Connection error" / "Connection closed"
+  // An odd number could result in a better length dispersion.
+  for (var i = 7; i <= 7*7*7; i *= 7)
+    headers.push(Array(i + 1).join('o'));
+
+  function WriteWithCRLF(line) {
+    todo.push(line + '\r\n');
+  }
+
+  function WriteHTTPHeaders(channel, has_keep_alive, extra_header_count) {
+    todo = []
+    WriteWithCRLF('GET / HTTP/1.1');
+    WriteWithCRLF('Host: localhost');
+    WriteWithCRLF('Connection: keep-alive');
+    WriteWithCRLF('Accept: text/html,application/xhtml+xml,' +
+                           'application/xml;q=0.9,image/webp,*/*;q=0.8');
+    WriteWithCRLF('User-Agent: Mozilla/5.0 (X11; Linux x86_64) ' +
+                           'AppleWebKit/537.36 (KHTML, like Gecko) ' +
+                           'Chrome/39.0.2171.71 Safari/537.36');
+    WriteWithCRLF('Accept-Encoding: gzip, deflate, sdch');
+    WriteWithCRLF('Accept-Language: en-US,en;q=0.8');
+    for (var i = 0; i < extra_header_count; i++) {
+      // Utilize first three powers of a small integer for an odd cycle and
+      // because the fourth power of some integers overloads the server.
+      WriteWithCRLF('X-Header-' + i + ': ' + headers[i%3]);
+    }
+    WriteWithCRLF('');
+    todo = todo.join('');
```

Comment:
I think you can replace WriteWithCRLF() with todo.push(), then do `todo = todo.join('\r\n')`?